docs/standard/microservices-architecture/implement-resilient-applications/monitor-app-health.md

```diff
@@ -15,146 +15,236 @@ In the typical model, services send reports about their status, and that informa
+                    var command = connection.CreateCommand();
+                    command.CommandText = TestQuery;
+
+                    await command.ExecuteNonQueryAsync(cancellationToken);
+                }
+            }
+            catch (DbException ex)
+            {
+                return new HealthCheckResult(status: context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        return HealthCheckResult.Healthy();
     }
 }
 ```
+Note in the above code, `Select 1` is the query used to check the Health of the database. To monitor the availability of your microservices, orchestrators like Kubernetes and Service Fabric periodically perform health checks by sending requests to test the microservices. It is important to keep your database queries very efficient so that these operations are quick and don’t result in a higher utilization of resources. 
 
-Thus, a microservice will not provide a “healthy” status until all its checks are healthy as well.
-
-If the microservice does not have a dependency on a service or on SQL Server, you should just add a Healthy("Ok") check. The following code is from the eShopOnContainers basket.api microservice. (The basket microservice uses the Redis cache, but the library does not yet include a Redis health check provider.)
+Finally, create a middleware that responds to the url path “/hc”.
 
 ```csharp
-services.AddHealthChecks(checks =>
+// Startup.cs from .NET Core 2.2 Web Api sample
+//
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 {
-    checks.AddValueTaskCheck("HTTP Endpoint", () => new
-        ValueTask<IHealthCheckResult>(HealthCheckResult.Healthy("Ok")));
-});
+    //…
+    app.UseHealthChecks("/hc");
+    //…
+} 
 ```
+When the endpoint `<yourmicroservice>/hc` is invoked, it runs all the health checks that are configured in the `AddHealthChecks()` method in the Startup class and shows the result.
+
+### HealthChecks implementation in eShopOnContainers
+
+Microservices in eShopOnContainers rely on multiple services to perform its task. For example, the Catalog.API microservice from eShopOnContainers depends on many services like Azure Blob Storage, SQL Server, RabbitMQ etc, therefore has several health checks added using the `AddCheck()` method. For every depending service, a custom `IHealthCheck` implementation that defines its respective health status needs to be added. 
+
+The open source project [AspNetCore.Diagnostics.HealthChecks](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks) solves this problem by providing custom health check implementations for each of these enterprise services that are built on top of .NET Core 2.2. Each health check is available as an individual Nuget package that can be easily added to the project. eShopOnContainers use them extensively in all its microservices.
+
+For instance, in the Catalog.API microservice, we added the following Nuget Packages:
+
+![](./media/image6.png)
```

Comment:
missing alt text 

we should really start using descriptive image file names for SEO reasons